pkg/server/restart.go

```diff
@@ -114,20 +115,26 @@ func (c *criService) recover(ctx context.Context) error {
 	// with best effort.
 
 	// Cleanup orphaned sandbox directories without corresponding containerd container.
-	if err := cleanupOrphanedSandboxDirs(sandboxes, filepath.Join(c.config.RootDir, "sandboxes")); err != nil {
+	if err := cleanupOrphanedSandboxDirs(sandboxes, filepath.Join(c.config.RootDir, sandboxesDir)); err != nil {
 		return errors.Wrap(err, "failed to cleanup orphaned sandbox directories")
 	}
+	if err := cleanupOrphanedSandboxDirs(sandboxes, filepath.Join(c.config.StateDir, sandboxesDir)); err != nil {
+		return errors.Wrap(err, "failed to cleanup orphaned volatile sandbox directories")
+	}
 
 	// Cleanup orphaned container directories without corresponding containerd container.
-	if err := cleanupOrphanedContainerDirs(containers, filepath.Join(c.config.RootDir, "containers")); err != nil {
+	if err := cleanupOrphanedContainerDirs(containers, filepath.Join(c.config.RootDir, containersDir)); err != nil {
 		return errors.Wrap(err, "failed to cleanup orphaned container directories")
 	}
+	if err := cleanupOrphanedContainerDirs(containers, filepath.Join(c.config.StateDir, containersDir)); err != nil {
```

Comment:
Nit: `cleanupOrphanedContainerDirs` and `cleanupOrphanedSandboxDirs` look very much alike.
Maybe it's worth combining them into `cleanupOrphanedSubdirs` or something.

Reply:
Will do.

Reply:
Done.

Reply:
Done.